plugins/woocommerce/src/Blocks/Utils/BlocksSharedState.php

```diff
@@ -94,11 +99,21 @@ public function register_cart_interactivity( $consent_statement ) {
 		self::check_consent( $consent_statement );
 
 		if ( null === self::$blocks_shared_cart_state ) {
-			$cart_exists                    = isset( WC()->cart );
-			$cart_has_contents              = $cart_exists && ! WC()->cart->is_empty();
-			self::$blocks_shared_cart_state = $cart_exists
-				? rest_do_request( new \WP_REST_Request( 'GET', '/wc/store/v1/cart' ) )->data
-				: array();
+			$cart_exists       = isset( WC()->cart );
+			$cart_has_contents = $cart_exists && ! WC()->cart->is_empty();
+			if ( isset( WC()->cart ) ) {
```

Comment:
Small nitpick that I will add by maybe just reuse the `$cart_exists`
```suggestion
			if ( $cart_exists ) {
```